onyo/commands/tests/test_unset.py

```diff
@@ -2,494 +2,383 @@
+    assert ret.stdout.count(f"-{key}") == len(repo.get_asset_paths())
     assert not ret.stderr
     assert ret.returncode == 0
-    fsck(repo)
+    assert repo.git.is_clean_worktree()
 
 
-@pytest.mark.repo_contents(*contents)
+@pytest.mark.repo_contents(*convert_contents([t for t in asset_contents if "num" in t[1]]))
 def test_unset_without_path(repo: OnyoRepo) -> None:
-    """
-    Test that `onyo unset --keys KEY` without a given path argument selects all
-    assets recursively.
-    """
-    key = list(content_dict.keys())[0]
+    """Test that `onyo unset --keys KEY` without a given path selects all assets recursively."""
+    key = "num"
     ret = subprocess.run(['onyo', '--yes', 'unset', '--keys', key],
                          capture_output=True, text=True)
 
     # verify the output
     assert "The following assets will be changed:" in ret.stdout
-    assert ret.stdout.count(f"-{key}: {content_dict.get(key)}") == len(assets)
+    assert ret.stdout.count(f"-{key}") == len(repo.get_asset_paths())
     assert not ret.stderr
     assert ret.returncode == 0
-    fsck(repo)
+    assert repo.git.is_clean_worktree()
 
 
-@pytest.mark.repo_contents(*contents)
-@pytest.mark.parametrize('directory', directories)
-def test_unset_recursive_directories(repo: OnyoRepo, directory: str) -> None:
-    """
-    Test that `onyo unset --keys KEY --path DIRECTORY` updates contents of
-    assets in DIRECTORY.
+@pytest.mark.repo_contents(*convert_contents([t for t in asset_contents if "num" in t[1]]))
+@pytest.mark.parametrize('directory', [Path('one/'), Path('another/')])
+def test_unset_recursive_directories(repo: OnyoRepo,
+                                     directory: Path) -> None:
+    """Test that `onyo unset --keys KEY --path DIRECTORY` updates contents of assets in DIRECTORY.
     """
-    key = list(content_dict.keys())[0]
-    ret = subprocess.run(['onyo', '--yes', 'unset', '--keys', key, '--path', directory], capture_output=True, text=True)
+    key = "num"
+    ret = subprocess.run(['onyo', '--yes', 'unset', '--keys', key, '--path', directory],
+                         capture_output=True, text=True)
 
     # verify changes, and the repository clean
-    # TODO: update this after solving #259
-    repo_assets = repo.asset_paths
-    for asset in [asset for asset in Path(directory).iterdir() if asset in repo_assets]:
-        assert key not in Path(asset).read_text()
-        assert f"-{key}: {content_dict.get(key)}" in ret.stdout
-    fsck(repo)
-
-
-@pytest.mark.repo_contents(*contents)
-@pytest.mark.parametrize('asset', assets)
-def test_unset_discard_changes_single_assets(repo: OnyoRepo, asset: str) -> None:
-    """
-    Test that `onyo unset` discards changes for assets successfully.
-    """
-    key = list(content_dict.keys())[0]
+    for asset in repo.get_asset_paths([directory]):
+        assert key not in asset.read_text()
+        assert f"-{key}" in ret.stdout
```

Comment:
Hm. I don't think this line does what you wanted it to do. It will always pass as long as the `-{key}` shows up once in the diff because you always look in `ret.stdout` as a whole not in something you iterate over. If you want to make sure that's the case per file, you need to count (like elsewhere).